src/Logging/InitializationLogger.cs

```diff
@@ -11,28 +11,33 @@ namespace Microsoft.Omex.Extensions.Logging
 	/// InitializationLogger is the logger to be used before the proper ILogger from DI is set.
 	/// Not to be used as main logger.
 	/// </summary>
-	[Obsolete($"{nameof(InitializationLogger)} using {nameof(OmexLogger)} is obsolete and is pending for removal by 1 July 2024.", DiagnosticId = "OMEX188")]
 	public static class InitializationLogger
 	{
 		/// <summary>
 		/// Instance of logger
 		/// </summary>
-		public static ILogger Instance { get; private set; } = LoggerFactory.Create(builder =>
-		 {
-			builder.LoadInitializationLogger();
-		 }).CreateLogger("Initial-Logging");
-
-		private static ILoggingBuilder LoadInitializationLogger(this ILoggingBuilder builder)
+		public static ILogger Instance
 		{
-			if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "Development")
-			{
-				builder.AddConsole();
-			}
+			get =>
+				LoggerFactory.Create(builder =>
+				{
+					if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "Development")
+					{
+						builder.AddConsole();
+					}
 
-			builder.AddOmexLogging();
-			return builder;
+					s_loggingBuilderAction?.Invoke(builder);
+				}).CreateLogger("Initial-Logging");
 		}
 
+		/// <summary>
+		/// Action to customize the initialization logger builder
+		/// </summary>
+		/// <param name="action"></param>
+		public static void CustomizeInitializationLoggerBuilder(Action<ILoggingBuilder> action) => s_loggingBuilderAction = action;
```

Comment:
It's not making life easier for consumers. Instead of wrapping host initialization in try/catch and explicitly logging failures, you want them to call this hook to customize the logger. And then they still have to add try/catch and logging to capture exceptions from IHost.Run[Async].